app/controllers/reports_controller.rb

```diff
@@ -351,7 +444,12 @@ def csv_data
       else
         csv << []
         csv << @headers
-        @data.each { |row| csv << row }
+        @data.each do |row|
```

Comment:
Grouped reports display arrays and references becuz of the new changes. This code fixes it:

```
@data.each do |group, pivot_table|
  csv << []
  csv << (@group_link && group.is_a?(Array) ? ["#{group[0]} #{group[1].room_number}"] : [group])
  csv << @headers
  pivot_table.each do |keys, record|
    keys = [keys[0][0]] + keys[1..] if @room_link && keys[0].is_a?(Array)
    csv << keys + @date_headers.map { |date| record[date] }
  end
end
```

Reply:
@Mohammad4844 Thanks, I forgot to check all CSV reports.